packages/application-extension/src/index.tsx

```diff
@@ -328,6 +331,32 @@ const mainCommands: JupyterFrontEndPlugin<void> = {
         isEnabled: () => !labShell.isEmpty('right')
       });
 
+      commands.addCommand(CommandIDs.activateSidebarWidget, {
+        label: trans.__('Open Sidebar Element'),
+        caption: trans.__('By default opens first element of left sidebar'),
+        execute: args => {
+          const index = parseInt(args.index as string, 10);
+          if (args.side != 'left' && args.side != 'right') {
+            throw Error(`Unsupported sidebar: ${args.side}`);
+          }
+          const widgets = Array.from(labShell.widgets(args.side));
+          if (index >= widgets.length) {
+            throw Error(`No item with index ${index} in ${args.side} sidebar`);
+          }
+          const widgetId = widgets[index].id;
+          // activate widget to show the associated content
+          labShell.activateById(widgetId);
```

Comment:
Shouldn't it be useful to toggle the panel instead of just activating it ?
e.g. toggle-filebrowser command

Reply:
Unsure of the indented change here

Not sure what the difference of Activating the widget (simulating an on click) and toggling the widget.
secondly it is unclear to me that all the tab bar widgets can be toggled and if its even possible to access it by just having the widgets id.

The intended functionality here is to have a keyboard shortcut for every widget on the tab bar that can be used from anywhere in the application to activate and more tab focus to that widget

Reply:
My idea was to be able to close an open widget.
It might be interesting to be able to close the sidebar using the same keyboard shortcut used to open it.

For example, I'm not sure if there is a shortcut to close the right panel.

Reply:
changes made, can you please review